player/lua/ytdl_hook.lua

```diff
@@ -57,6 +59,35 @@ local function edl_escape(url)
     return "%" .. string.len(url) .. "%" .. url
 end
 
+local function time_to_secs(time_string)
+    if not time_string then
+        return nil
+    end
+
+    local secs = 0
+    string.gsub(time_string, "[%d.]+", function (number)
+        secs = secs * 60 + tonumber(number)
+    end)
+
+    return secs
+end
+
+local function extract_chapters(data, video_length)
+    local ret = {}
+
+    for line in data:gmatch("[^\r\n]+") do
+        local time_string = string.match(line, "([%d:.]+:[%d.]+)")
```

Comment:
Nit: it's best to make the second `%d` be the optional one (`%d%d?:`) so there's no need to backtrack on, e.g. `1:23`.

With `%d?%d:`: `%d?` matches, `%d` doesn't; backtrack and try with `%d?` not matching, `%d` matches, `:` matches.

With `%d%d?`: `%d` matches, `%d?` doesn't, `:` matches.

As for functions, I'd be partial to `istime2` just because it's smaller and based on an algorithm that's already used in `mpv`, but if they pass tests either works.